src/sagas/helpers.js

```diff
@@ -175,3 +176,47 @@ export async function getRegisteredTokens(wallet, excludeHTR = false) {
 
   return tokens;
 }
+
+export async function getFullnodeNetwork() {
+  // NOTE: Should we change the user-agent?
+  // config.setUserAgent('wallet-mobile');
+  try {
+    const versionData = await new Promise((resolve) => {
+      hathorLib.versionApi.getVersion(resolve);
+    });
+
+    return versionData.network;
+  } catch {
+    throw new Error('Error getting fullnode version data.');
+  }
+}
+
+export async function getWalletServiceNetwork() {
+  // NOTE: Should we change the user-agent?
+  // config.setUserAgent('wallet-mobile');
+  try {
+    const timeout = WALLET_SERVICE_REQUEST_TIMEOUT;
+    // eslint-disable-next-line max-len
+    const instance = axiosWrapperCreateRequestInstance(config.getWalletServiceBaseUrl, null, timeout);
+    const response = await instance.get(`version`);
+    return response.data.network;
+  } catch {
+    throw new Error('Error getting wallet-service version data.');
+  }
+}
+
+export function isWalletServiceUnavailable(networkSettings) {
+  return networkSettings.walletServiceUrl == null;
+}
+
+export function disableFeaturesIfNeeded(networkSettings, currentFeatureToggles) {
+  let featureToggles = { ...currentFeatureToggles };
+  if (isWalletServiceUnavailable(networkSettings)) {
```

Comment:
I improved by adding docstrings at 5ba3e501c77cea9dc42389b122b5916bb19d2fe4

But cutting short, `networkSettings` is the new custom network settings and this function verifies if the wallet service URL is not there, if yes "it is not there" we need to forcefully disable the features that depends on the wallet-service.

This whole process is just to enforce the right feature toggles, because it is updated regularly by a polling process.